src/dns_internal.cc

```diff
@@ -211,20 +211,34 @@ class ConfigRr : public RegisteredRunner
     void endingShutdown() override;
 };
 
-} // namespace Dns
-
-DefineRunnerRegistratorIn(Dns, ConfigRr);
+/// a Top-Level Domain (TLD) to use when looking up an unqualified hostname
+class SearchTld
+{
+public:
+    explicit SearchTld(const char *buf) {
+        xstrncpy(domain, buf, sizeof(domain));
+        Tolower(domain);
+    }
 
-struct _sp {
     char domain[NS_MAXDNAME];
```

Comment:
while touching this, would it make sense to migrate domain to SBuf?

Reply:
Aye, SBuf conversion is deferred until the code using this has been through more refactoring cleanup.